### DISCUSSION TITLE
Button click tracking with WebAppInfo

### DISCUSSION THREAD

**Discussion by jekeam** [python-telegram-bot/python-telegram-bot#3333] [Q&A] [answered]
How can I track user click on button with WebAppInfo? (I do not have access to the called application)
```
KeyboardButton(
    text='some text',
    web_app=WebAppInfo(url='some_url')
)
```

**Comment by Poolitzer** [comment, marked as answer]
Define track? If you want to have a user object when opening the website, use an inline keyboard button instead.

**Reply by jekeam** [reply]
> Define track? If you want to have a user object when opening the website, use an inline keyboard button instead.

I can't use the built in keyboard. i am using ReplyKeyboardMarkup, here is my code, how can i track the user click event on the button?
```
main_markup = ReplyKeyboardMarkup(
    [[
        KeyboardButton(
              text='some text',
              web_app=WebAppInfo(url='some_url')
    )],],
    resize_keyboard=True,
    one_time_keyboard=False,
)
```

**Reply by Poolitzer** [reply]
You can't, use an inline keyboard if you want to see which user opens the webpage.

**Comment by coolsnake** [comment]
I use the following code,but I don't know how to track the "open_website" button click for web_app. Any one can help me? Thanks.
keyboard = [
        [
            InlineKeyboardButton(lang[LANGUAGE]["settings"], callback_data="language"),
            InlineKeyboardButton(lang[LANGUAGE]["open_website"], 
                                 web_app=WebAppInfo(url=f"{MINI_APP_URL}/{LANGUAGE}"))
        ],
    ]

**Reply by jekeam** [reply]
There is no such technical possibility at the moment
